radio_beam/beam.py

```diff
@@ -27,6 +27,16 @@ def _to_area(major,minor):
                u.arcmin: "'"}
 
 
+def _with_default_unit(type_str, value, unit, equiv_unit=u.deg):
+    if not hasattr(value, 'unit'):
+        return value * unit
+
+    if value.unit.is_equivalent(equiv_unit):
```

Comment:
I think your input `unit` argument is always going to be equivalent to `equiv_unit`, right? (Realistically, here, both will be angular units.) If so, one might as well use `value.unit.is_equivalent(unit)` here  (and thus remove the `equiv_unit` argument).

Reply:
`value` can be passed with any unit, and this is the first place where it checks against `equiv_unit`, while also accounting for not having enforced `value` to be a Quantity (which would make sense but alters the default behaviour in previous radio-beam version). 

I think it would hit a units error later on, but not necessarily in `Beam.__init__`.

Reply:
I think you misunderstood my point: if, e.g., the arguments are `unit=u.arcsec, equiv_unit=u.degree`, i.e., when the two unit arguments are equivalent between themselves, then `value.unit.is_equivalent(unit)` is guaranteed to do exactly the same as `value.unit.is_equivalent(equiv_unit)`. So, I don't think there is a need for the `equiv_unit` argument, you can just do the test with the `unit` argument here.

Reply:
Ah I see your point! Thanks @mhvk! Fixed.